src/utils/get_tx_data.ts

```diff
@@ -0,0 +1,38 @@
+import Arweave from 'arweave';
+import axios, { AxiosInstance } from 'axios';
+import axiosRetry, { exponentialDelay } from 'axios-retry';
+import { ArFSMetadataCache } from '../arfs/arfs_metadata_cache';
+import { gatewayUrlForArweave, TransactionID } from '../exports';
+
+export async function getDataForTxID(
+	txId: TransactionID,
+	arweave: Arweave,
+	axiosInstance?: AxiosInstance // Provided for test mocking :(
```

Comment:
It would be a goal :soccer: if we could set up global mocking properly!

Reply:
BTW we could have a default here and get rid of the line #19 above (`axiosInstance ??= axios.create();`)
```suggestion
	axiosInstance: AxiosInstance = axios.create() // Provided for test mocking :(
```

Reply:
2ecc89c